plugins/quickinspector/quickoverlay.cpp

```diff
@@ -50,6 +50,50 @@ extern Q_GUI_EXPORT QImage qt_gl_read_framebuffer(const QSize &size,
 QT_END_NAMESPACE
 
 namespace GammaRay {
+
+class QQuickItemPropertyCache {
+public:
+    static const QQuickItemPropertyCache &getPropertyCache(QQuickItem *item)
+    {
+        static QHash<QByteArray, QQuickItemPropertyCache> s_cache;
+        const auto meta = item->metaObject();
+        const auto className = QByteArray(meta->className());
```

Comment:
QByteArray(const char*) make a deep copy of the data.
I would use QHash<QLatin1String, QQuickItemPropertyCache> s_cache and QLatin1String(mo->className()).
And track item deletion to remove it from the hash.